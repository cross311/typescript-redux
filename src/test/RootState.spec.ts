import { combineReducers } from "redux";
import {
  Action,
  buildRootState,
  combineRootStateReducers,
  EntityCollection,
  entityCollectionInitialState,
  EntitySingle,
  entitySingleInitialState,
  initialState,
  RootStateReducers,
} from "../";

describe("RootState", () => {
  it("should build root state", () => {
    const base: RootStateReducers = {
      singles: { base: (state: EntitySingle<{}> = entitySingleInitialState(), action: Action<{}>) => state },
      collections: { base: (state: EntityCollection<{}> = entityCollectionInitialState(), action: Action<{}>) => state },
      pages: { base: (state: {} = {}, action: Action<{}>) => state },
    };

    const otherReducers: RootStateReducers = {
      pages: {
        base: (state: {} = {}, action: Action<{}>) => state,
      },
    };

    const rootState = buildRootState(combineRootStateReducers([
      base,
      otherReducers,
    ]));

    expect(rootState).toMatchSnapshot();
  });

  it("should give an empty initial state", () => {
    expect(initialState()).toEqual({
      entities: {
        singles: {},
        collections: {},
      },
      pages: {},
    });
  });
});
