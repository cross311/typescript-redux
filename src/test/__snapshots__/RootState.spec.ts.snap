// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootState should build root state 1`] = `
Object {
  "entities": [Function],
  "pages": [Function],
}
`;
